### PR TITLE
Update SDK for new deployment spec

### PR DESCRIPTION
**Pull request by paf31**
- Use new environment variable names and defaults
- Remove configuration server
- Pass `PathBuf` to `validate_raw_configuration` and remove `RawConfiguration`
- Update `Dockerfile` to meet the deployment spec

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,6 +16,10 @@ COPY ./rust-connector-sdk .
 RUN cargo build --release
 
 FROM debian:buster-slim as connector
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive \
+    apt-get install --no-install-recommends --assume-yes\
+    libssl-dev
```
Comment (SamirTalwar) [inline Dockerfile:22]: Adding `set -ex` makes it easier to figure out which line broke when something goes wrong.

```suggestion
RUN set -ex; \
    apt-get update; \
    DEBIAN_FRONTEND=noninteractive \
      apt-get install --no-install-recommends --assume-yes \
      libssl-dev
```

File: rust-connector-sdk/src/connector.rs [hunk 2]
```diff
@@ -201,23 +191,15 @@ pub enum MutationError {
...
-
-    async fn update_configuration(
-        config: Self::RawConfiguration,
-    ) -> Result<Self::RawConfiguration, UpdateConfigurationError>;
-
     /// Validate the raw configuration provided by the user,
     /// returning a configuration error or a validated [`Connector::Configuration`].
     async fn validate_raw_configuration(
-        configuration: Self::RawConfiguration,
+        configuration_dir: PathBuf,
```
Comment (SamirTalwar) [inline rust-connector-sdk/src/connector.rs:202]: I suggest weakening this to `AsRef<Path>` so we can be a little more flexible with the input.

```suggestion
        configuration_dir: impl AsRef<Path>,
```

File: rust-connector-sdk/src/default_main.rs [hunk 3]
```diff
@@ -243,17 +233,13 @@ where
 
 /// Initialize the server state from the configuration file.
 pub async fn init_server_state<C: Connector + Clone + Default + 'static>(
-    config_file: impl AsRef<Path>,
+    config_directory: PathBuf,
```
Comment (SamirTalwar) [inline rust-connector-sdk/src/default_main.rs:236]:
```suggestion
    config_directory: impl AsRef<Path>,
```

**Review by SamirTalwar — APPROVED** [review]
Approving so it's not blocked, but please consider my comments.

File: rust-connector-sdk/src/default_main.rs (hunk 3 above)
Comment (paf31) [reply rust-connector-sdk/src/default_main.rs:236]: It would have to be `AsRef<Path> + Send` apparently.

Comment (SamirTalwar) [reply rust-connector-sdk/src/default_main.rs]: Fair; that's OK with me if it's OK with you. Looks like that's fine for `PathBuf` and `String`.

**Review by SamirTalwar — APPROVED** [review]

File: rust-connector-sdk/src/connector.rs [hunk 4]
```diff
@@ -201,23 +191,15 @@ pub enum MutationError {
...
 
-    fn make_empty_configuration() -> Self::RawConfiguration;
-
-    async fn update_configuration(
-        config: Self::RawConfiguration,
-    ) -> Result<Self::RawConfiguration, UpdateConfigurationError>;
-
     /// Validate the raw configuration provided by the user,
     /// returning a configuration error or a validated [`Connector::Configuration`].
     async fn validate_raw_configuration(
```
Comment (SamirTalwar) [inline rust-connector-sdk/src/connector.rs:201]: While we're here, how about renaming this to `parse_configuration`?

File: rust-connector-sdk/src/default_main.rs [hunk 5]
```diff
@@ -60,69 +53,68 @@ enum Command {
...
 #[derive(Clone, Parser)]
 struct TestCommand {
     #[arg(long, value_name = "SEED", env = "SEED")]
     seed: Option<String>,
-    #[arg(long, value_name = "CONFIGURATION_FILE", env = "CONFIGURATION_FILE")]
+    #[arg(
+        long,
+        value_name = "DIRECTORY",
+        env = "HASURA_CONFIGURATION_DIRECTORY",
+        default_value = "/etc/connector"
```
Comment (daniel-chambers) [inline rust-connector-sdk/src/default_main.rs:88]: I'm not sure how easy this is to do, but is it possible to remove this default when compiled for Windows? I know we don't care about deployment on Windows, but theoretically Windows developers may build a connector using this SDK and when they run it locally, it should not default to an impossible path, instead we should prefer to just make them specify this manually.

Comment (daniel-chambers) [reply rust-connector-sdk/src/default_main.rs:88]: This also applies to the other commands that have this same parameter: ServeCommand, ReplayCommand

Comment (SamirTalwar) [reply rust-connector-sdk/src/default_main.rs:88]: Hmmm, I think this point is bigger than just Windows vs. Unices.

In a Linux Docker image, we want to read `/etc/connector`. But when running locally for testing, we want to read the specified directory; we almost never want to use `/etc/connector`, even on Linux.

I think the right move here is to remove the default value and set `ENV HASURA_CONFIGURATION_DIRECTORY /etc/connector` in connector Docker images. This can be overridden.

Long story short, the default is a packaging concern, not an SDK concern.

Comment (paf31) [reply rust-connector-sdk/src/default_main.rs:88]: As I think about it, I agree. I'll remove the default and move it to the Dockerfile. For reference, it was possible though:

```rust
#[cfg_attr(target_family = "unix", arg(default_value = "/etc/connector"))]
```

**Review by SamirTalwar — APPROVED** [review]

**Comment by SamirTalwar** [comment]
Let's merge this. Gonna follow up with a change to an error enum but that needs its own review.
